cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,545 @@
+            this.selectedFidelityPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($fidelityContainer);
+            this.populateLanguages();
+        },
+
+        turnaroundSelected: function(event) {
+            var $turnaroundContainer = this.$el.find('.transcript-turnaround-wrapper');
+
+            this.selectedTurnaroundPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($turnaroundContainer);
+            this.populateLanguages();
+        },
+
+        providerSelected: function(event) {
+            this.resetPlanData();
+            this.selectedProvider = event.target.value;
+            this.populatePreferences();
+        },
+
+        languageAdded: function(event) {
+            var $parentEl = $(event.target.parentElement).parent(),
+                $languagesEl = this.$el.find('.transcript-languages-wrapper'),
+                selectedLanguage = $parentEl.find('select').val();
+
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($languagesEl);
+
+            // Only add if not in the list already.
+            if (selectedLanguage && _.indexOf(this.selectedLanguages, selectedLanguage) === -1) {
+                this.selectedLanguages.push(selectedLanguage);
+                HtmlUtils.setHtml(
+                    $parentEl,
+                    HtmlUtils.joinHtml(
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<span>{languageDisplayName}</span>'),
+                            {
+                                languageDisplayName: this.availableLanguages[selectedLanguage]
+                            }
+                        ),
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<div class="remove-language-action"><button class="button-link action-remove-language" data-language-code="{languageCode}">{text}<span class="sr">{srText}</span></button></div>'),
+                            {
+                                languageCode: selectedLanguage,
+                                text: gettext('Remove'),
+                                srText: gettext('Press Remove to remove language')
+                            }
+                        )
+                    )
+                );
+
+                // Add a new language menu
+                this.addLanguageMenu();
+            } else {
+                this.addErrorState($languagesEl);
+            }
+        },
+
+        languageRemoved: function(event) {
+            var selectedLanguage = $(event.target).data('language-code');
+            $(event.target.parentElement).parent().remove();
+            this.selectedLanguages.pop(selectedLanguage);
+        },
+
+        populateProvider: function() {
```

Comment:
I did this to keep things dynamic if in future we get a new provider, it can easily fit in. @muzaffaryousaf Do you think, we need to make this static?